tests/monitoring/component_monitoring.go

```diff
@@ -162,108 +228,173 @@ var _ = Describe("[sig-monitoring]Component Monitoring", Serial, decorators.SigM
 		BeforeEach(func() {
 			virtClient = kubevirt.Client()
 
+			By("Backing up the operator cluster role binding")
 			crb, err = virtClient.RbacV1().ClusterRoleBindings().Get(context.Background(), "kubevirt-operator", metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
+
+			By("Backing up the operator role binding")
 			operatorRoleBinding, err = virtClient.RbacV1().RoleBindings(flags.KubeVirtInstallNamespace).Get(context.Background(), operatorRoleBindingName, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
+			By("Increasing the rate limit")
 			increaseRateLimit()
 
 			scales = libmonitoring.NewScaling(virtClient, []string{virtOperator.deploymentName})
 			scales.UpdateScale(virtOperator.deploymentName, int32(0))
 
+			By("Reducing the alert pending time")
 			libmonitoring.ReduceAlertPendingTime(virtClient)
 		})
 
 		AfterEach(func() {
+			By("Restoring the operator cluster role binding")
 			crb.Annotations = nil
 			crb.ObjectMeta.ResourceVersion = ""
 			crb.ObjectMeta.UID = ""
 			_, err = virtClient.RbacV1().ClusterRoleBindings().Create(context.Background(), crb, metav1.CreateOptions{})
 			Expect(err).To(Or(Not(HaveOccurred()), MatchError(errors.IsAlreadyExists, "IsAlreadyExists")))
 
+			By("Restoring the operator role binding")
 			operatorRoleBinding.Annotations = nil
 			operatorRoleBinding.ObjectMeta.ResourceVersion = ""
 			operatorRoleBinding.ObjectMeta.UID = ""
 			_, err = virtClient.RbacV1().RoleBindings(flags.KubeVirtInstallNamespace).Create(context.Background(), operatorRoleBinding, metav1.CreateOptions{})
 			Expect(err).To(Or(Not(HaveOccurred()), MatchError(errors.IsAlreadyExists, "IsAlreadyExists")))
-			scales.RestoreAllScales()
 
-			waitUntilComponentsAlertsDoNotExist(virtClient)
+			By("Restoring the operator scale")
+			oldVirtOperatorReplicas := scales.GetScale(virtOperator.deploymentName)
+
+			for i := int32(1); i <= oldVirtOperatorReplicas; i++ {
+				By(fmt.Sprintf("Updating the operator scale to %d", i))
+				scales.UpdateScale(virtOperator.deploymentName, i)
+
+				By("Waiting for the operator to be up")
+				libmonitoring.WaitForMetricValue(virtClient, "kubevirt_virt_operator_up", float64(i))
+
+				By("Waiting for the operator to be ready")
+				libmonitoring.WaitForMetricValue(virtClient, "kubevirt_virt_operator_ready", float64(i))
+			}
+
+			By("Waiting for an operator to be leading")
+			libmonitoring.WaitForMetricValue(virtClient, "kubevirt_virt_operator_leading", 1.0)
 		})
 
-		It("VirtApiRESTErrorsBurst and VirtApiRESTErrorsHigh should be triggered when requests to virt-api are failing", func() {
+		It("VirtApiRESTErrorsBurst should be triggered when requests to virt-api are failing", func() {
```

Comment:
why not in all alerts tests we use WaitForMetricValue first?

Reply:
All the tests in "Errors metrics" don't have it, I found that in these tests it doesn't impact the flakiness of the tests